File: 2025/56. Merge Intervals/Solution.java
class Solution {
    public int[][] merge(int[][] intervals) {
        if (intervals == null || intervals.length < 0)
            return intervals;
        if (intervals.length < 2)
            return intervals;
        List<int[]> list = new ArrayList<>();
        Arrays.sort(intervals, (int[] x, int[] y) -> (x[0] - y[0]));
        int i = 0, j = 1, start = intervals[0][0], end = intervals[0][1];
        while (j < intervals.length) {
            if (intervals[j][0] > end) {
                list.add(new int[]{start, end});
                start = intervals[j][0];
                end = intervals[j][1];
            } else {
                end = Math.max(intervals[j][1], end);
            }
            j++;
        }
        list.add(new int[]{start, end});
        return list.toArray(new int[list.size()][2]);
    }
}

File: 2025/56. Merge Intervals/solution.py
class Solution:
    def merge(self, intervals: List[List[int]]) -> List[List[int]]:
        intervals = sorted(intervals, key=lambda x: (x[0]))
        res, prev = [], intervals[0]
        for cur in intervals:
            if cur[0] > prev[1]:
                res.append([prev[0], prev[1]])
                prev = cur
            else:
                prev[1] = max(cur[1], prev[1])
        res.append(prev)
        return res